store/watcher_hub.go

```diff
@@ -69,13 +73,16 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index, storeInde
 		hub:        wh,
 	}
 
-	wh.mutex.Lock()
```

Comment:
I moved it up to before the scan, that's where my priority inversion question came from.
I moved it up there because I don't want an event to creep in after I've scanned but before I've added this watcher (we can still add an event, but watcher notification is blocked on the mutex).